tests/unit/generateNotificationData.unitTest.ts

```diff
@@ -1,54 +1,58 @@
+                const waitActivitiesArray = JSON.parse(waitActivitiesList.body);
+                const sendNotificationData = notificationData.generateActivityDetails(visit, sendAtfReport.computeActivitiesList(testResultsArray, waitActivitiesArray));
+                expect(sendNotificationData.testStationPNumber).toEqual(testResultsArray[0].testStationPNumber);
+                expect(sendNotificationData.testerName).toEqual(visit.testerName);
+                expect(sendNotificationData.startTimeDate).toEqual("14/01/2019");
+                expect(sendNotificationData.startTime).toEqual("08:47:33");
+                expect(sendNotificationData.endTime).toEqual("15:36:33");
+                expect(sendNotificationData.activityDetails.length).not.toEqual(0);
             });
         });
 
         context("when parsing the visit and multiple test results", () => {
             it("should return a correct test stations emails with dividers", () => {
-                const testResultsArray = JSON.parse(testResultsListMultiTest.body);
-                const sendNotificationData = notificationData.generateActivityDetails(visit, testResultsArray);
-                expect(sendNotificationData.testStationPNumber).to.equal(testResultsArray[0].testStationPNumber);
-                expect(sendNotificationData.testerName).to.equal(visit.testerName);
-                expect(sendNotificationData.startTimeDate).to.equal("14/01/2019");
-                expect(sendNotificationData.startTime).to.equal("08:47:33");
-                expect(sendNotificationData.endTime).to.equal("15:36:33");
-                expect(sendNotificationData.activityType).to.equal("Test");
-                expect(sendNotificationData.activityDetails.length).to.not.equal(0);
-                expect(countInstances(sendNotificationData.activityDetails,"---")).to.equal(2);
+                const testResultsArray = TestResultsService.prototype.expandTestResults(JSON.parse(testResultsListMultiTest.body));
+                const sendNotificationData = notificationData.generateActivityDetails(visit, sendAtfReport.computeActivitiesList(testResultsArray, []));
+                expect(sendNotificationData.testStationPNumber).toEqual(testResultsArray[0].testStationPNumber);
+                expect(sendNotificationData.testerName).toEqual(visit.testerName);
+                expect(sendNotificationData.startTimeDate).toEqual("14/01/2019");
+                expect(sendNotificationData.startTime).toEqual("08:47:33");
+                expect(sendNotificationData.endTime).toEqual("15:36:33");
+                expect(sendNotificationData.activityDetails.length).not.toEqual(0);
+                expect(countInstances(sendNotificationData.activityDetails,"---")).toEqual(2);
             });
         });
+
+        // TODO Add tests to verify the waitTime fields for "Time not Testing" activity added in ATF report.
+        // Verify for activityType, startTime, endTime, waitReason values.
```

Comment:
I see that the above tests include wait activities. Is this TODO still valid?

Reply:
replaced with test